qiskit/circuit/library/standard_gates/h.py

```diff
@@ -222,3 +225,6 @@ def _define(self):
     def inverse(self):
         """Return inverted CH gate (itself)."""
         return CHGate(ctrl_state=self.ctrl_state)  # self-inverse
+
+    def __eq__(self, other):
+        return isinstance(other, CHGate) and self.ctrl_state == other.ctrl_state
```

Comment:
Same comment. Perhaps a shared implementation is possible for all `SingletonControlledGate` instances?